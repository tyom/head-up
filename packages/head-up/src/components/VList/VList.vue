<template>
  <div class="VList" :class="listClass">
    <ul>
      <li
        v-for="(item, idx) in items"
        :key="idx"
      >
        <slot :data="item"/>
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    type: {
      type: String,
      default: 'stack',
    },
    padded: {
      type: Boolean,
      default: false,
    },
    items: {
      type: Array,
      default: () => [],
    },
  },
  computed: {
    listClass() {
      return {
        _padded: this.padded,
        [`_${this.type}`]: this.type,
      };
    },
  },
};
</script>

<style scoped>
.VList {
  height: 100%;
}

ul {
  height: 100%;
  list-style: none;
  padding: 0;
  text-align: left;
}

li {
  background-color: rgba(#fff, 0.02);

  & + li {
    margin-top: 2px;
  }
}

._padded > ul > li {
  padding: 0.5em;
}

._tiles > ul {
  display: grid;
  grid-gap: 0.5em;
  grid-template-columns: repeat(auto-fill, minmax(20em, 1fr));

  & > li {
    margin-top: 0;
    display: flex;
  }
}
</style>
