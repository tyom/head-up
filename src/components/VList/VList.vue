<template>
  <div class="VList" :class="listClass">
    <ul>
      <li v-for="(item, idx) in items" :key="idx">
        <slot :item="item" />
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    type: {
      type: String,
      default: 'stack',
    },
    padded: {
      type: Boolean,
      default: false,
    },
    items: {
      type: Array,
      default: () => [],
    },
  },
  computed: {
    listClass() {
      return {
        _padded: this.padded,
        [`_${this.type}`]: this.type,
      };
    },
  },
};
</script>

<style scoped>
.VList {
  height: 100%;
}

ul {
  height: 100%;
  list-style: none;
  padding: 0;
  text-align: left;
}

li {
  @apply shadow;
  background-color: #fff1;
}

._padded > ul > li {
  padding: 0.5em;
}

._tiles > ul {
  display: grid;
  grid-gap: 0.5em;
  grid-template-columns: repeat(auto-fill, minmax(25rem, 1fr));
  grid-template-rows: repeat(auto-fill, minmax(25rem, 1fr));

  & > li {
    margin-top: 0;
    display: flex;
  }
}
</style>
