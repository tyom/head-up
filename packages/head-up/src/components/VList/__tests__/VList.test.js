import { shallowMount } from '@vue/test-utils';
import VList from '../VList';

test('render default', () => {
  const wrapper = shallowMount(VList);
  expect(wrapper).toMatchSnapshot();
});

test('render with items', () => {
  const wrapper = shallowMount(VList, {
    propsData: {
      items: [{ label: 'One' }, { label: 'Two' }],
    },
    scopedSlots: {
      default: '<div slot-scope="{item}">{{ item.label }}</div>',
    },
  });
  expect(wrapper).toMatchSnapshot();
});
