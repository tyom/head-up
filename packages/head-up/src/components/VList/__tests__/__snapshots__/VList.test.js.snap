// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render default 1`] = `
<div class="VList">
  <ul class="_stack"></ul>
</div>
`;

exports[`render with items 1`] = `
<div class="VList">
  <ul class="_stack">
    <li>
      <div>One</div>
    </li>
    <li>
      <div>Two</div>
    </li>
  </ul>
</div>
`;
